OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -368,11 +373,61 @@ ActorInfo ChooseBuildingToBuild(ProductionQueue queue)
 			return null;
 		}
 
+		static CVec[] Directions(int distance)
+		{
+			return new CVec[]
+			{
+				new CVec(-distance, -distance),
+				new CVec(-distance,  0),
+				new CVec(-distance,  distance),
+				new CVec(0, -distance),
+				new CVec(0,  distance),
+				new CVec(distance, -distance),
+				new CVec(distance,  0),
+				new CVec(distance,  distance),
+			};
+		}
+
+		bool TooCloseToBuilding(CPos cell, ActorInfo actorInfo, BuildingInfo buildingInfo, BaseBuilderBotModule baseBuilder)
+		{
+			if (baseBuilder.Info.BaseSpacing == null)
+				return false;
+
+			if (!baseBuilder.Info.BaseSpacing.ContainsKey(actorInfo.Name))
+				return false;
+
+			var distance = baseBuilder.Info.BaseSpacing[actorInfo.Name];
```

Comment:
Consider using `baseBuilder.Info.BaseSpacing.TryGetValue` to avoid two lookups.

Reply:
Good idea. I applied this on multiple occasions.